Basic styling improvements

mor4thii opened the pull request.

mor4thii committed:
🐛 Set subtitle to being h2 instead of h1 #8

mor4thii committed:
💄 Style social bar and add legal notice icon #8

mor4thii committed:
💄 Make vertical containers centered #8

mor4thii committed:
💄 Style the PageTitle component #8

doppelmutzi commented on src/components/Link.tsx:
```diff
@@ -18,7 +20,7 @@ interface LinkProps {
 export const Link = ({ url, label, icon }: LinkProps) => (
-    <Anker href={url} textLink={!icon}>
-        {icon}
+    <Anker href={url} textLink={!icon} target="_blank">
```
anker -> anchor

doppelmutzi replied:
spelling -> anchor

doppelmutzi commented on src/components/SocialBar.tsx:
```diff
@@ -1,17 +1,42 @@
+import { faGavel } from '@fortawesome/free-solid-svg-icons'
+import {
+    FontAwesomeIcon,
+    FontAwesomeIconProps,
```
unused code

doppelmutzi commented on src/components/SocialBar.tsx:
```diff
@@ -1,17 +1,42 @@
+import { colors } from '../styles/theme'
 import { Link } from './Link'
 
+type SocialBarItemPros = {
```
Pros -> Props

doppelmutzi replied:
we should either use types or interfaces for prop definitions.

mor4thii replied:
I used interfaces as they better fit the purpose!

doppelmutzi commented on src/components/Subtitle.tsx:
```diff
@@ -1,4 +1,13 @@
+    <LSubtitle>{children}</LSubtitle>
+)
+
+const LSubtitle = styled.h2`
```
this are no layouting styles. So don't use the "L" prefix

doppelmutzi commented on src/components/Headline.tsx:
```diff
@@ -1,4 +1,13 @@
+    <LHeadline>{children}</LHeadline>
+)
+
+const LHeadline = styled.h1`
```
this are no layouting styles. So don't use the "L" prefix

doppelmutzi reviewed (dismissed):
Good job! Only minor issues.

mor4thii committed:
💄 Introduce a website-wide font (#8)

mor4thii committed:
♻️ Remove the "L" prefix from some styles (#8)

mor4thii committed:
✏️ Use the english word for "anchor"

mor4thii committed:
🔥 Remove an unused import

mor4thii committed:
♻️ Use interfaces where we define objects (#8)

Types only allow intersect types, interfaces allow clear extension
through "extends". The type keyword is rather for type aliases, unions,
or tuples.

doppelmutzi commented on src/styles/resetCSS.ts:
```diff
@@ -1,6 +1,20 @@
 
+import MikiRegular from '../fonts/Miki/KontrapunktMiki-Regular.otf'
+
 const GlobalStyle = createGlobalStyle`
```
Hätte ich jetzt nicht hier erwartet. Als Global Style sollte richtig sein, so dass es in jeder React Component greift.

mor4thii replied:
Fixed

doppelmutzi commented on src/components/Subtitle.tsx:
```diff
@@ -1,4 +1,13 @@
 import React from 'react'
```
Kann man das ohne diesen Import nutzen? Falls nicht, dann müssen wir noch was an unserem TS/Webpack setup anpassen?!

mor4thii replied:
Das geht irgendwie, siehe https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html
Aber ich möchte jetzt erst mal nicht noch babel-support zum Projekt hinzufügen aktuell. Lass uns das in einer dedizierten Story machen!

mor4thii committed:
♻️  Refactor global font into global style file

mor4thii commented:
@doppelmutzi, ich habe den globalStyle change eingebaut, aber nicht das mit dem import statement. Du kannst gerne noch einmal drüberschauen, bitte.

doppelmutzi commented:
Ich reviewe nochmal am Samstag. 

Bzgl react import. Bei dm hab ich das setup hingekriegt. Ich gugg da auch mal drauf.

PS probiere gerade mal per smartphone das hier zu bedienen.
